section_to_course/admin.py

```diff
@@ -0,0 +1,357 @@
+        if course_exists(key):
+            raise ValidationError(
+                _('A course with this number, org, and run already exists. Please choose different values.')
+            )
+        return self.cleaned_data
+
+    def save(self, *args, **kwargs):
+        """
+        Create the course and then copy the section into it.
+        """
+        cleaned_data = self.cleaned_data
+        org = cleaned_data['new_course_org']
+        number = cleaned_data['new_course_number']
+        run = cleaned_data['new_course_run']
+        course = create_course(
+            user=self.user,
+            org=org,
+            number=number,
+            run=run,
+            display_name=cleaned_data['new_course_name'],
+        )
+        return paste_from_template(
+            destination_course_key=course.id,
+            source_block_usage_key=cleaned_data['source_section_id'],
+            user=self.user,
+        )
+
+    class Meta:
+        """
+        Meta configuration for SectionToCourse model form.
+        """
+
+        fields = (
+            'source_course_id',
+            'source_section_id',
+            'new_course_name',
+            'new_course_org',
+            'new_course_number',
+            'new_course_run',
+        )
+        model = SectionToCourseLink
+
+
+@admin.action(description=_('Refresh section content from source.'))
+def refresh_courses(model_admin, request, queryset):
+    """Refresh selected courses in the admin."""
+    for item in queryset:
+        paste_from_template(
+            destination_course_key=item.destination_course_id,
+            source_block_usage_key=item.source_section_id,
+            user=request.user,
+        )
+    model_admin.message_user(request, _('Refreshed {} courses successfully.').format(queryset.count()))
+
+
+class SectionToCourseLinkAdmin(DjangoObjectActions, admin.ModelAdmin):
+    """
+    Admin view for section to course links.
+    """
+
+    list_display = ('name', 'source_course_id', 'source_section_id', 'destination_course_id', 'last_refresh', 'link')
+    list_filter = ('source_course_id', 'destination_course_id')
+    actions = [refresh_courses]
+    change_actions = ('refresh_this', )
+
+    def refresh_this(self, request, obj):
+        """
+        Refresh this course from its source via a special button on the edit page.
+        """
+        paste_from_template(
+            destination_course_key=obj.destination_course_id,
+            source_block_usage_key=obj.source_section_id,
+            user=request.user,
+        )
+        self.message_user(request, _("Refreshed course successfully."))
+
+    refresh_this.label = _("Refresh Course Content")
+    refresh_this.short_description = _("Refresh this course's content from the source section.")
+
+    def name(self, obj):  # pylint: disable=no-self-use
+        """
+        Display course name.
+        """
+        dest_course = get_course(obj.destination_course_id)
```

Comment:
I quickly checked, and `get_course_outline(obj.destination_course_id)` (from the Learning Sequences) is still about 3x faster for me locally. We can change this in a follow-up, though.